src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingtoolsapi/controller/DefinitionController.kt

```diff
@@ -10,30 +10,30 @@ import org.springframework.web.bind.annotation.PutMapping
 import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.digitalprisonreportinglib.data.model.ProductDefinition
-import uk.gov.justice.digital.hmpps.digitalprisonreportingtoolsapi.data.InMemoryProductDefinitionRepository
+import uk.gov.justice.digital.hmpps.digitalprisonreportingtoolsapi.service.DefinitionService
 
 @RestController
 @Tag(name = "Report Definition API")
-class DefinitionsController(val repository: InMemoryProductDefinitionRepository) {
+class DefinitionController(val definitionService: DefinitionService) {
   @Operation(
     description = "Saves a definition",
     security = [SecurityRequirement(name = "bearer-jwt")],
   )
-  @PutMapping("/definitions")
+  @PutMapping("/definition")
   fun putDefinition(
     @RequestBody
     @Valid
     definition: ProductDefinition,
   ) {
-    repository.save(definition)
+    definitionService.validateAndSave(definition)
   }
 
   @Operation(
     description = "Deletes a definition",
     security = [ SecurityRequirement(name = "bearer-jwt") ],
   )
-  @DeleteMapping("/definitions/{definitionId}")
+  @DeleteMapping("/definition/{definitionId}")
```

Comment:
I know we have discussed this but here is another suggestion as an option.
We could have PUT /definitions/{definitionId} and DELETE /definition/{definitionId}.
In that way definitions refers to the collection of definitions and it is consistent in both PUT and DELETE.